firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -266,7 +268,8 @@ class SearchViewController: SiteTableViewController,
         return Task { [weak self] in
             guard let suggestions = try? await self?.profile.firefoxSuggest?.query(
                 tempSearchQuery,
-                providers: providers
+                providers: providers,
+                limit: self?.maxNumOfFirefoxSuggestions
```

Comment:
what do we think about checking if self is nil first on? Therefore, we can avoid making limit param as an optional type since we're always passing in a value that is not nil.

Reply:
I like this suggestion! 🙌🏼

@tiftran You're almost there! You can use a `guard let`, like this, to unwrap the limit:

```swift
return Task { [weak self] in
    guard let limit = self?.maxNumOfFirefoxSuggestions,
          let suggestions = try? await self?.profile.firefoxSuggest?.query(
            tempSearchQuery,
            providers: providers,
            limit: limit
    ) else { return }
    // ...
}
```